src/server/hook_func.c

```diff
@@ -5321,6 +5328,38 @@ mk_deferred_hook_info(int index, int event, long long int tid)
 	return info;
 }
 
+/**
+ * @brief
+ *		check if there is any new pending action
+ *		if yes remove the current one
+ *	
+ * @param[in] minfo - pointer to mom info
+ * @param[in] pact - pointer to current hook action
+ * @param[in] j - index of pact in minfo->mi_action[]
+ * @param[in] event - action event to consider
+ *
+ * @return int
+ * @retval 0  - not found
+ * @retval 1  - found
+ */
+int
+check_for_latest_action(mominfo_t *minfo, mom_hook_action_t *pact, int j, int event)
+{
+	/* if the same action is marked in the next actions for the same hook then remove
+	* the pending flag
+	*/
+	int i;
+	mom_hook_action_t *pact2;
+	for (i = 0; i < minfo->mi_num_action; i++) {
+		pact2 = minfo->mi_action[i];
+		if (pact2 && (pact2->tid > pact->tid) && (pact2->action & event)
+				&& pact2->hookname && (strcmp(pact2->hookname, pact->hookname) == 0)) {
```

Comment:
Better readability with the && in the previous line

Reply:
Done https://github.com/openpbs/openpbs/pull/1992/commits/d54ce17afb4e9d6cd1a3da070504726722e60e5c